net/dnsfallback/dnsfallback.go

```diff
@@ -143,3 +177,58 @@ func getDERPMap() *tailcfg.DERPMap {
 
 //go:embed dns-fallback-servers.json
 var staticDERPMapJSON []byte
+
+// cachedDERPMap is the path to a cached DERP map that we loaded from our on-disk cache
+var cachedDERPMap syncs.AtomicValue[*tailcfg.DERPMap]
+
+// cachePath is the path to the DERP map cache file
+var cachePath string
+
+// UpdateCache stores the DERP map cache back to disk.
+func UpdateCache(c *tailcfg.DERPMap) {
```

Comment:
I guess we take ownership of c? Or the caller must promise to not mutate it at least?